keps/sig-cli/3104-introduce-kuberc/README.md

```diff
@@ -0,0 +1,849 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This proposal introduces an optional `kuberc` file that is used to separate cluster credentials and server configuration from user preferences. 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+kubectl is one of the oldest parts of the Kubernetes project and has a strong guarantee on backwards compatibility. We want users to be able to opt in to new behaviors (like delete confirmation) that may be considered breaking changes for existing CI jobs and scripts. 
+
+[kubeconfigs already have a field for preferences](https://github.com/kubernetes/kubernetes/blob/474fd40e38bc4e7f968f7f6dbb741b7783b0740b/staging/src/k8s.io/client-go/tools/clientcmd/api/types.go#L43) that is currently underutilized. The reason for not using this existing field is that creating a new cluster generally yields a new kubeconfig file which contains credentials and host information. While kubeconfigs can be merged and specified by path, we feel there should be a clear separation between server configuration and user preferences.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Introduce a `kuberc` file as a place for client preferences.
+* Version and structure this file in a way that makes it easy to introduce new behaviors and settings for users.
+* Enable users to define kubectl command aliases.
+* Enable users to define kubectl default flags.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We propose introducing a new file that will be versioned and intended for user provided preferences. This new file is entirely opt-in.
```

Comment:
Currently, besides defining contexts, .kube/config also stores the user's currently selected context...
```
current-context: cluster1
```

Do you see that remaining in .kube/config or will it be moving here?

Reply:
I think that's a good idea but will have the same issue as the XDG change. Any tooling that hasn't been updated could use a different context.

Maybe we could reject any value in kuberc if `current-context` is defined in the kubeconfig?

Thoughts?

Reply:
Probably something we'll need to figure out at alpha, add that to questions section, or elements to figure out before beta.